### ISSUE TITLE
Duplicate key in Edit as YAML

### ISSUE THREAD

**Issue by ryansann** [rancher/dashboard#4136]
**Setup**
- Rancher version: v2.6.0 / 2.6-head
- Browser type & version: Chrome 93.0

**Describe the bug**
When editing as YAML from a eeployment config, the UI adds a duplicate key in the YAML

**To Reproduce**
1. Create a simple deployment
2. Navigate to the deployment and select `Edit Config`
3. Once inside the form based editor, choose `Edit as YAML` at the bottom of the form
4. Note that saving will not work until duplicate key is removed

**Result**
Duplicate key is created as seen here:
```yaml
  strategy:
    type: RollingUpdate
    rollingUpdate:
      maxSurge: 25%
      maxUnavailable: 25%
    type: RollingUpdate
```
`type` occurs twice in the same object

**Expected Result**
Should be given valid YAML when switching from edit config to edit as yaml

**Comment by nwmac** [comment]
@rak-phillip Looks like the same as the nodePort issue

**Comment by rak-phillip** [comment]
@nwmac definitely looks to be the same as #4042. I'll test the fix being reviewed in #4132 to make sure that this case is covered.

**Comment by sowmyav27** [comment]
**Reproduced the issue on 2.6.0**
<img width="677" alt="Screen Shot 2021-09-17 at 4 24 42 PM" src="https://user-images.githubusercontent.com/26032343/133863281-63ddd034-7b61-4d23-8d5a-068111d60058.png">

**On 2.6-head commit id: `1a57856`**
<img width="678" alt="Screen Shot 2021-09-17 at 4 25 19 PM" src="https://user-images.githubusercontent.com/26032343/133863295-efdbc222-2181-4d97-8ea4-1b44166cdc58.png">
